pastas_plugins/modflow/modflow.py

```diff
@@ -162,16 +154,18 @@ def update_model(self, p: ArrayLike):
 
         self._gwf.name_file.write()
 
+    @functools.lru_cache(maxsize=5)
+    def _get_head(self, p):
+        self.update_model(p=p)
+        success, _ = self._simulation.run_simulation(silent=True)
+        if success:
+            return self._gwf.output.head().get_ts((0, 0, 0))[:, 1]
+        else:
+            raise Exception("Model run failed")
```

Comment:
If the model does not converge during optimization we now have an issue / exception. That's why the zeroes were returned.

Reply:
Fair point, I'll make it an option. Perhaps something we also want to report :).